### PR TITLE
Cleanup of Near Cache related classes

### PR DESCRIPTION
**Pull request by Donnerbart**
* Made `ClientExecutionService` implement `ExecutionService`
* Removed obsolete `NearCacheExecutor`
* Unified `initNearCache()` method names
* Moved `NearCacheContext` creation to `NearCacheProvider` constructor

Please review together with the linked Hazelcast Enterprise PR.

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/ClientNearCacheExecutor.java [hunk 1]
```diff
@@ -24,14 +24,14 @@
 
 public final class ClientNearCacheExecutor implements NearCacheExecutor {
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/ClientNearCacheExecutor.java]: Imho this class is just noise. Define a single ScheduledExecutor interface with scheduleWithRepetition and offer it on a 'util' level. Then we don't need to make all these wrapper classes.

File: hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java [hunk 2]
```diff
@@ -79,10 +79,10 @@ public NearCachedClientMapProxy(String serviceName, String name) {
     protected void onInitialize() {
         super.onInitialize();
 
-        init();
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java]: What is the extra value of calling this 'initNearCache'?

File: hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/MemberNearCacheExecutor.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+package com.hazelcast.map.impl.nearcache;
+
+import com.hazelcast.cache.impl.nearcache.NearCacheExecutor;
+import com.hazelcast.spi.ExecutionService;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public final class MemberNearCacheExecutor implements NearCacheExecutor {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/MemberNearCacheExecutor.java]: Another piece of noise.

File: hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java (hunk 2 above)
Comment (Donnerbart) [inline hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java]: It's overridden in Hazelcast EE.

**Comment by Donnerbart** [comment]
I extended the cleanup to get rid of the `NearCacheExecutor` by adding `ExecutionService` to the client implementation. Maybe the client and member implementation could be unified or more abstracted, but this is beyond the scope of this PR. This PR is limited to cleanup the usage in the Near Cache, which is already improved a lot by getting rid of the useless wrapper class and instances.

File: hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/ClientNearCacheExecutor.java [hunk 4]
```diff
@@ -1,37 +0,0 @@
...
-
-package com.hazelcast.client.cache.impl;
-
-import com.hazelcast.cache.impl.nearcache.NearCacheExecutor;
-import com.hazelcast.client.spi.ClientExecutionService;
-
-import java.util.concurrent.ScheduledFuture;
-import java.util.concurrent.TimeUnit;
-
-public final class ClientNearCacheExecutor implements NearCacheExecutor {
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/ClientNearCacheExecutor.java]: Deleting code always make me happy.

File: hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java [hunk 5]
```diff
@@ -20,14 +20,18 @@
 import com.hazelcast.core.ICompletableFuture;
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java]: Not part of this PR; but is this class needed at all? 

So why do we need all these wrapper executors in the first place.

**Comment by pveentjer** [comment]
:+1: 

But another pair of eyes is needed

File: hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java (hunk 5 above)
Comment (Donnerbart) [inline hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java]: Don't know exactly yet. The client also implements `Executor` and has an "internal" and "external" executor, the member has some stuff to register and retrieve named executors. So there are some extensions, but maybe we can unify those classes in the end, even if some functionality is just used in either member or clients.

File: hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientExecutionService.java [hunk 6]
```diff
@@ -17,19 +17,18 @@
...
-import java.util.concurrent.ScheduledFuture;
-import java.util.concurrent.TimeUnit;
 
 /**
  * Executor service for Hazelcast clients.
  *
  * Allows asynchronous execution and scheduling of {@link Runnable} and {@link Callable} commands.
  */
-public interface ClientExecutionService extends Executor {
+public interface ClientExecutionService extends ExecutionService, Executor {
```
Comment (ahmetmircik) [inline hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientExecutionService.java]: 👍
